templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -36,8 +35,8 @@ contents:
                 done
             fi
     EOF
-        # Ensure resolv.conf exists before we try to run podman
-        if [[ ! -e /etc/resolv.conf ]]; then
+        # Ensure resolv.conf exists and contains nameservers before we try to run podman
+        if [[ ! -e /etc/resolv.conf || -z $(grep -q nameserver /etc/resolv.conf) ]]; then
```

Comment:
Could check the return code from grep directly by doing this:

if [[ ! -e /etc/resolv.conf ]] || ! grep -q nameserver /etc/resolv.conf; then